EngineLayer/Silac/SilacConversions.cs

```diff
@@ -332,267 +269,602 @@ public static SpectraFileInfo GetHeavyFileInfo(SpectraFileInfo originalFile, Sil
-                            for (int j = 1; j < identifications.Count; j++) //add all the original identification
+                            //else if three values, then we're fine
+                            //add the values from this peptide to the file-specific-values of all peptides
+                            double[] totalValues = fileToRecycleDictionary[info];
+                            for (int i = 0; i < values.Count; i++)
                             {
-                                updatedPeak.MergeFeatureWith(new FlashLFQ.ChromatographicPeak(identifications[j], peak.IsMbrPeak, peak.SpectraFileInfo), Integrate);
+                                totalValues[i] += values[i];
                             }
-                            updatedPeak.IsotopicEnvelopes = peak.IsotopicEnvelopes; //need to set isotopicEnevelopes, since the new identifications didn't have them.
-                            updatedPeak.CalculateIntensityForThisFeature(Integrate); //needed to update info
-                            peaks[i] = updatedPeak;
                         }
                     }
-                }
 
-                //convert all lfq peptides from heavy to light for output
-                Dictionary<string, FlashLFQ.Peptide> lfqPwsms = FlashLfqResults.PeptideModifiedSequences;
-                List<string> pwsmKeys = lfqPwsms.Keys.ToList();
-                foreach (string key in pwsmKeys)
-                {
-                    FlashLFQ.Peptide currentPeptide = lfqPwsms[key];
-                    SilacLabel label = GetRelevantLabelFromFullSequence(currentPeptide.Sequence, silacLabels);
-                    if (label != null) //if it's a heavy peptide
+                    //with the summed LL, LH, and HH values for each file, we can calculate the probability that a heavy amino acid is incorporated (Ph)
+                    //the probability a light amino acid is incorporated (Pl) is simply Pl = 1 - Ph
+                    //We can simplify the math if we assume that all protein started as completely light, @t=0: LL=100%, LH = 0%, and HH = 0%
+                    //N = 1 - Qll + Qlh * Qlh / (4 * Qhh), (where N is the fraction of all protein that has turned over and Qll is the Quantity(fraction) of LL at the current timepoint
+                    //Qhh = qhh -N*qhh+Ph*Ph*N, where Q is quantity(fraction) at the timepoint and q is the quantity(fraction) at the start point (time zero).
+                    //Ph = sqrt((Qhh-qhh+N*qhh)/N), and qhh = 0, so we can simplify as
+                    //Ph = sqrt(Qhh/N)
+                    Dictionary<SpectraFileInfo, double> fileToHeavyProbabilityDictionary = new Dictionary<SpectraFileInfo, double>();
+                    foreach (SpectraFileInfo info in spectraFileInfo)
+                    {
+                        //convert the absolute values of LL, LH, and HH to relative values
```

Comment:
nice to have a unit test to see if the probabilities used are correct.

Reply:
Full task unit tests are testing that the probabilities were calculated and applied correctly.